pkg/scheduler/algorithmprovider/registry.go

```diff
@@ -50,17 +50,23 @@ const ClusterAutoscalerProvider = "ClusterAutoscalerProvider"
 // Registry is a collection of all available algorithm providers.
 type Registry map[string]*schedulerapi.Plugins
 
-// NewRegistry returns an algorithm provider registry instance.
-func NewRegistry() Registry {
-	defaultConfig := getDefaultConfig()
-	applyFeatureGates(defaultConfig)
+// ApplyFeatureGates applies feature gates to registry
+func (r Registry) ApplyFeatureGates() Registry {
+	if defaultConfig, ok := r[schedulerapi.SchedulerDefaultProviderName]; ok {
+		applyFeatureGates(defaultConfig)
+	}
+	if caConfig, ok := r[ClusterAutoscalerProvider]; ok {
+		applyFeatureGates(caConfig)
+	}
 
-	caConfig := getClusterAutoscalerConfig()
-	applyFeatureGates(caConfig)
+	return r
```

Comment:
I don't think this should return `r`. Wouldn't it be simpler to make `ApplyFeatureGates` just be a function without a return value?

also, I think this is treating the symptom rather than the root cause; shouldn't we just add a verbosity filter to the `klog.Infof` that is polluting stdout?

Reply:
I think, don't return `r` just make it more complex like:
```
r := NewRegistry()
r.ApplyFeatureGates()
```
Why not just `r.NewRegistry().ApplyFeatureGates()`?